src/GraphQL.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Netzstrategen\WooCommerceLocalStore;
+
+/**
+ * GraphQL related functionalities.
+ */
+class GraphQL {
+
+  /**
+   * @implements graphql_register_types
+   */
+  public static function graphql_register_types() {
+    self::registerLocalStockStatusEnum();
+    self::registerLocalStockObjectType();
+    self::registerLocalStockField('product');
+    self::registerLocalStockField('ProductVariation');
+  }
+
+  /**
+   * Registers local stock status enum.
+   */
+  public static function registerLocalStockStatusEnum() {
+    $values = [
+      'HIGH' => ['value' => 'high'],
+      'LOW' => ['value' => 'low'],
+      'NONE' => ['value' => 'none'],
+    ];
+
+    register_graphql_enum_type(
+    'LocalStockStatus',
+    [
+      'description' => __('Local stock status enumeration', Plugin::L10N),
+      'values' => $values,
+    ]
+    );
+  }
+
+  /**
+   * Registers localStock object type.
+   */
+  public static function registerLocalStockObjectType() {
+    register_graphql_object_type(
+      'localStock',
+      [
+        'description' => __('Local stock', Plugin::L10N),
+        'fields' => [
+          'storeName' => [
+            'type' => 'String',
+            'description' => __('Store name', Plugin::L10N),
+          ],
+          'showroom' => [
+            'type' => 'LocalStockStatus',
+            'description' => __('Showroom', Plugin::L10N),
+          ],
+          'warehouse' => [
+            'type' => 'LocalStockStatus',
+            'description' => __('Warehouse', Plugin::L10N),
+          ],
+          'online' => [
+            'type' => 'LocalStockStatus',
+            'description' => __('Online', Plugin::L10N),
+          ],
```

Comment:
Not every store/location has a showroom or warehouse.  Only the online stock applies to all.

Here's the store configuration of GACO: https://github.com/netzstrategen/gaco/blob/master/.woocommerce-local-store.json

For completeness, maybe we should also expose the internal store name as `name` and rename the current `storeName` into `title` or `label`?

For example:
```js
name: lauchringen
label: Lauchringen
showroom: NONE
warehouse: LOW
online: HIGH
```

Reply:
1. 

I am using in the `resolver` the function `Product::getStockByStore($product)`. It is returning the following structure 

```
array(3) {
  ["Stockach"]=>
  array(3) {
    ["showroom"]=>
    string(4) "none"
    ["warehouse"]=>
    string(4) "none"
    ["online"]=>
    string(4) "high"
  }
  ["Lauchringen"]=>
  array(3) {
    ["showroom"]=>
    string(4) "none"
    ["warehouse"]=>
    string(4) "none"
    ["online"]=>
    string(4) "high"
  }
  ["Weil am Rhein"]=>
  array(3) {
    ["showroom"]=>
    string(4) "none"
    ["warehouse"]=>
    string(4) "none"
    ["online"]=>
    string(4) "high"
  }
}
```

2. 
I am using the `key ` as the store name, so I don't get the same structure as seen in https://github.com/netzstrategen/gaco/blob/master/.woocommerce-local-store.json, namely the internal store name.

It is okay if a store doesn't have a particular facility (warehouse, showroom, online), we are showing them on the table with `'none'`. We are basically resolving the data the same way the plugin has been doing it for the front end.

Reply:
Ah, that makes sense!  I forgot that we're only keeping the label for the theme output.  I guess it's fine to pass the label in the GraphQL data now - if we stumble over problems, then we can still refactor.

However, I'd still name the property "label" then - it signifies that it is a human-readable title, and also, that the value is not stable and it might change (e.g. translated store labels in the future).